shefParser

```diff
@@ -2899,20 +2925,52 @@ class ShefParser :
                 interval_value = int(token[3:])
                 if abs(interval_value) > 99 :
                     raise ShefParser.ParseException(f"Invalid interval value: [{token}]")
+                duration_code = None
                 if interval_unit == 'S' :
                     interval = timedelta(seconds=interval_value)
+                    try :
+                        duration_code = [x for x in self._duration_codes if self._duration_codes[x] == interval_value + 7000][0]
```

Comment:
Could this be a function , and recycled?

def lookup_duration_code(offset)

Reply:
Modified to a single lookup `duration_id = self._duration_ids[duration_code]`